app/views/records/edit_fields/_creator.html.erb

```diff
@@ -2,46 +2,56 @@
 <% creator_types = HykuAddons::NameTypeService.new.select_active_options.flatten.uniq! %>
 
 <% f.object.creator_list.each do |hash| %>
-	<div class="ubiquity-meta-creator">
-		<div class="form-group">
-			<% required = template != 'collection' %>
-
-			<label class="control-label multi_value optional" for="<%= template %>_creator_name_type">
-				Creator name type
-				<% if required %>
-					<span class="label label-info required-tag">required</span>
-				<% end %>
-			</label>
+	<div class="js-select-toggle">
+		<% field_options = {
+			field_slug: :creator_name_type,
+			select_options: creator_types,
+			required: template != "collection",
+			field_args: { class: 'js-select-toggle-change form-control' },
+			field_group: :creator,
+			template: template,
+			hash: hash
+		} %>
+		<%= render "records/edit_fields/attribution_partials/select_field", field_options %>
 
```

Comment:
Love this generalisation! I wonder if we could take this a bit further and make the views slightly different?
If we create a method with the following signature:

```
def render_deposit_form_field(field_slug, field_type = :text, options = {})
  options[:required] ||= false
  ...
  render "records/edit_fields/attribution_partials/#{field_type}_field", options
end
```
Then we can put the following in the views:

<%= render_deposit_form_field :creator_organisation_name %>
<%= render_deposit_form_field :creator_name_type, :collection, required: true, field_group: '...' %>  

What do you think guys?